### ISSUE TITLE
DatabaseTypeName is always empty

### ISSUE THREAD

**Issue by xtrimf** [vertica/vertica-sql-go#77]
why the **DatabaseTypeName** is always empty?
https://golang.org/pkg/database/sql/#ColumnType.DatabaseTypeName

```Go
func Test_vertica(t *testing.T) {
	connDB, err := sql.Open("vertica", "vertica://user:password@localhost:5433/test")
	if err != nil {
		fmt.Println(err)
	}
	defer connDB.Close()
	ctx := context.Background()
	err = connDB.PingContext(ctx)
	rows, err := connDB.QueryContext(ctx, "SELECT * FROM v_monitor.cpu_usage LIMIT 5")
	defer rows.Close()
	cols, _ := rows.ColumnTypes()
	spew.Dump(cols)
	for _, column := range cols {
		spew.Dump(column.DatabaseTypeName())
	}
}
```

Results:
```Log
=== RUN   Test_vertica
([]*sql.ColumnType) (len=4 cap=4) {
 (*sql.ColumnType)(0xc0000fc550)({
  name: (string) (len=9) "node_name",
  hasNullable: (bool) false,
  hasLength: (bool) false,
  hasPrecisionScale: (bool) false,
  nullable: (bool) false,
  length: (int64) 0,
  databaseType: (string) "",
  precision: (int64) 0,
  scale: (int64) 0,
  scanType: (*reflect.rtype)(0x67ea60)(interface {})
 }),
 (*sql.ColumnType)(0xc0000fc5a0)({
  name: (string) (len=10) "start_time",
  hasNullable: (bool) false,
  hasLength: (bool) false,
  hasPrecisionScale: (bool) false,
  nullable: (bool) false,
  length: (int64) 0,
  databaseType: (string) "",
  precision: (int64) 0,
  scale: (int64) 0,
  scanType: (*reflect.rtype)(0x67ea60)(interface {})
 }),
 (*sql.ColumnType)(0xc0000fc5f0)({
  name: (string) (len=8) "end_time",
  hasNullable: (bool) false,
  hasLength: (bool) false,
  hasPrecisionScale: (bool) false,
  nullable: (bool) false,
  length: (int64) 0,
  databaseType: (string) "",
  precision: (int64) 0,
  scale: (int64) 0,
  scanType: (*reflect.rtype)(0x67ea60)(interface {})
 }),
 (*sql.ColumnType)(0xc0000fc640)({
  name: (string) (len=25) "average_cpu_usage_percent",
  hasNullable: (bool) false,
  hasLength: (bool) false,
  hasPrecisionScale: (bool) false,
  nullable: (bool) false,
  length: (int64) 0,
  databaseType: (string) "",
  precision: (int64) 0,
  scale: (int64) 0,
  scanType: (*reflect.rtype)(0x67ea60)(interface {})
 })
}
(string) ""
(string) ""
(string) ""
(string) ""
--- PASS: Test_vertica (0.46s)
PASS
ok      command-line-arguments  (cached)
```

Tested with several tables (not only system tables) - same results
same for **ScanType**: https://golang.org/pkg/database/sql/#ColumnType.ScanType

**Comment by huebnerr** [comment]
I suspect it's not implemented. I'll have a look.

**Comment by huebnerr** [comment]
@xtrimf I have the name returning in my test branch correctly now as well as a few other fields. However, I want to make all of these completely accurate so am researching the mapping of internal column specs to these values. If all you need is the name, you can test out my branch. I'll do a PR once the research is complete.

**Comment by xtrimf** [comment]
type names are looking good!
still missing :
 ```
 precision: (int64) 0,
  scale: (int64) 0,
  scanType: (*reflect.rtype)(0xafe8a0)(interface {})
```
precision & scale for numeric column type
in scanType should show the go type (%T) , but its showing interface for everything

**Comment by xtrimf** [comment]
@huebnerr there is an issue for 'date' type:
```
(*sql.ColumnType)(0xc000140820)({
  name: (string) (len=14) "partition_date",
  hasNullable: (bool) false,
  hasLength: (bool) true,
  hasPrecisionScale: (bool) false,
  nullable: (bool) false,
  length: (int64) 8,
  databaseType: (string) (len=27) "**unknown column type oid: 10**",
  precision: (int64) 0,
  scale: (int64) 0,
  scanType: (*reflect.rtype)(0xafe8a0)(interface {})
 })
```

**Comment by xtrimf** [comment]
Hi, @huebnerr do you have an ETA for the PR?

**Comment by sitingren** [comment]
@xtrimf I have fixed DatabaseTypeName and Nullable, I'll fix DecimalSize, Length and ScanType in another PR.
